addon/active-model-serializer.js

```diff
@@ -292,6 +292,9 @@ var ActiveModelSerializer = RESTSerializer.extend({
           payload = hash[payloadKey];
         }
 
+        if (!payload) {
+          return;
+        }
```

Comment:
I'm unsure if this is the correct solution. `payload` might be `null` and that should be allowed iirc?

We should probably copy the `if (!hash.hasOwnProperty(payloadKey)) { return; }` from the non-polymorphic code path above into the polymorphic one (once we have the `payloadKey`).